azurelinuxagent/common/utils/extensionprocessutil.py

```diff
@@ -68,12 +68,13 @@ def handle_process_completion(process, command, timeout, stdout, stderr, error_c
     process_output = read_output(stdout, stderr)
 
     if timed_out:
-        raise ExtensionError("Timeout({0}): {1}\n{2}".format(timeout, command, process_output),
-                             code=ExtensionErrorCodes.PluginHandlerScriptTimedout)
+        raise LaunchCommandError("Timeout({0}): {1}\n{2}".format(timeout, command, process_output),
+                                 code=ExtensionErrorCodes.PluginHandlerScriptTimedout,
+                                 exit_code=ExtensionErrorCodes.PluginHandlerScriptTimedout)
 
     if return_code != 0:
-        raise ExtensionError("Non-zero exit code: {0}, {1}\n{2}".format(return_code, command, process_output),
-                             code=error_code)
+        raise LaunchCommandError("Non-zero exit code: {0}, {1}\n{2}".format(return_code, command, process_output),
```

Comment:
Need to change `ExtensionError` in `cgroupapi.py#start_extension_command` too?

Reply:
Thanks for catching it, totally missed this one! I was under the impression that we capture output only in 1 place!

Reply:
I see. My comment above still applies, though :)

Reply:
Totally agree to it, will modify the PR